extensions/azurePublishNew/src/components/provisioningWizards/HandOffToAdminWizard.tsx

```diff
@@ -4,26 +4,76 @@
+import { useRecoilValue } from 'recoil';
+import { ProvisionHandoff } from '@bfc/ui-shared';
 
 import { WizardStep, Wizard } from '../shared/wizard';
+import { useDispatcher } from '../../hooks/useDispatcher';
+import { enabledHandOffResourcesState } from '../../recoilModel/atoms/handOffToAdminState';
+import {
+  subscriptionState,
+  hostNameState,
+  resourceGroupState,
+  deployLocationState,
+} from '../../recoilModel/atoms/resourceConfigurationState';
 
 import { HandOffInstructionsStep } from './steps/HandOffInstructionsStep';
 import { WizardFooter } from './footers/WizardFooter';
+import { ChooseResourcesStep } from './steps/ChooseResourcesStep';
 
 type Props = {
-  onStepChange?: (stepIndex: number, stepId: string) => void;
+  onStepChange?: (stepIndex: number, step: WizardStep) => void;
   stepIndex?: number;
 };
 
+const urls = {
+  createNewResources: 'https://aka.ms/composer-publish-bot#create-new-azure-resources',
+};
+
 export const HandOffToAdminWizard = React.memo((props: Props) => {
   const { onStepChange } = props;
   const [steps, setSteps] = React.useState<WizardStep[]>([]);
-  const { setTitle, onBack, closeDialog: onCancel } = usePublishApi();
+  const { onBack, closeDialog: onCancel } = usePublishApi();
+  const resources = useRecoilValue(enabledHandOffResourcesState);
+  const subscriptionId = useRecoilValue(subscriptionState);
+  const hostName = useRecoilValue(hostNameState);
+  const deployRegion = useRecoilValue(deployLocationState);
+  const { name: resourceGroupName } = useRecoilValue(resourceGroupState);
+  const { setEnabledHandOffResources } = useDispatcher();
+  const [showHandOff, setShowHandOff] = React.useState<boolean>(false);
+
+  const handOffInstructions = React.useMemo(() => {
+    let instructions = '';
+    if (showHandOff) {
+      const createLuisResource = resources?.filter((r) => r.key === 'luisPrediction').length > 0;
+      const createLuisAuthoringResource = resources?.filter((r) => r.key === 'luisAuthoring').length > 0;
+      const createCosmosDb = resources?.filter((r) => r.key === 'cosmosDb').length > 0;
+      const createStorage = resources?.filter((r) => r.key === 'blobStorage').length > 0;
+      const createAppInsights = resources?.filter((r) => r.key === 'applicationInsights').length > 0;
+      const createQnAResource = resources?.filter((r) => r.key === 'qna').length > 0;
+
+      const provisionComposer = `node provisionComposer.js --subscriptionId ${
+        subscriptionId || '<YOUR SUBSCRIPTION ID>'
+      } --name ${hostName || '<RESOURCE NAME>'} --appPassword=<16 CHAR PASSWORD> --location=${
+        deployRegion || 'westus'
+      } --resourceGroup=${
+        resourceGroupName || '<RESOURCE GROUP NAME>'
+      } --createLuisResource=${createLuisResource} --createLuisAuthoringResource=${createLuisAuthoringResource} --createCosmosDb=${createCosmosDb} --createStorage=${createStorage} --createAppInsights=${createAppInsights} --createQnAResource=${createQnAResource}`;
```

Comment:
You can encapsulate this long logic in a stand alone hook:

useSomething(resources: type) => command:string

Reply:
good idea!!